SonarQube.Common/VerbosityCalculator.cs

```diff
@@ -0,0 +1,109 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="DefaultSettings.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+
+namespace SonarQube.Common
+{
+    public static class VerbosityCalculator
+    {
+        /// <summary>
+        /// The initial verbosity of loggers
+        /// </summary>
+        /// <remarks>
+        /// Each of the executables (bootstrapper, pre/post processor) have to parse their command line and config files before
+        /// being able to deduce the correct verbosity. In doing that it makes sense to set the logger to be more verbose before
+        /// a value can be set so as not to miss out on messages.
+        /// </remarks>
+        public const LoggerVerbosity InitialLoggingVerbosity = LoggerVerbosity.Debug;
+
+        /// <summary>
+        /// The verbosity of loggers if no settings have been specified
+        /// </summary>
+        public const LoggerVerbosity DefaultLoggingVerbosity = LoggerVerbosity.Info;
+
+        /// <summary>
+        /// Computes verbosity based on the available properties. 
+        /// </summary>
+        /// <remarks>If no verbosity setting is present, the default verbosity (info) is used</remarks>
+        public static LoggerVerbosity ComputeVerbosity(IAnalysisPropertyProvider properties, ILogger logger)
+        {
+            if (properties == null)
+            {
+                throw new ArgumentNullException("properties");
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            string sonarVerboseValue;
+            properties.TryGetValue(SonarProperties.Verbose, out sonarVerboseValue);
+
+            string sonarLogLevelValue;
+            properties.TryGetValue(SonarProperties.LogLevel, out sonarLogLevelValue);
+
+            return ComputeVerbosity(sonarVerboseValue, sonarLogLevelValue, logger);
+        }
+
+        /// <summary>
+        /// Computes verbosity based on the analysis config file settings. 
+        /// </summary>
+        /// <remarks>If no verbosity setting is present, the default verbosity (info) is used</remarks>
+        public static LoggerVerbosity ComputeVerbosity(AnalysisConfig config, ILogger logger)
+        {
+            if (config == null)
+            {
+                throw new ArgumentNullException("config");
+            }
+
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+
+            AnalysisSetting sonarVerboseSetting;
+            config.TryGetSetting(SonarProperties.Verbose, out sonarVerboseSetting);
```

Comment:
ConfigSettingsExtensions has a GetSetting(...) method that returns a default if the value is missing.